ext/ffi/ffi.c

```diff
@@ -2870,30 +2868,17 @@ static ZEND_FUNCTION(ffi_trampoline) /* {{{ */
 }
 /* }}} */
 
-static zend_function *zend_ffi_get_func(zend_object **obj, zend_string *name, const zval *key) /* {{{ */
+static zend_function *zend_ffi_scope_get_func(zend_object **obj, zend_string *name, const zval *key) /* {{{ */
 {
 	zend_ffi        *ffi = (zend_ffi*)*obj;
 	zend_ffi_symbol *sym = NULL;
 	zend_function   *func;
 	zend_ffi_type   *type;
+	zval *zv;
 
-	if (ZSTR_LEN(name) == sizeof("new") -1
-	 && (ZSTR_VAL(name)[0] == 'n' || ZSTR_VAL(name)[0] == 'N')
-	 && (ZSTR_VAL(name)[1] == 'e' || ZSTR_VAL(name)[1] == 'E')
-	 && (ZSTR_VAL(name)[2] == 'w' || ZSTR_VAL(name)[2] == 'W')) {
-		return (zend_function*)&zend_ffi_new_fn;
-	} else if (ZSTR_LEN(name) == sizeof("cast") -1
-	 && (ZSTR_VAL(name)[0] == 'c' || ZSTR_VAL(name)[0] == 'C')
-	 && (ZSTR_VAL(name)[1] == 'a' || ZSTR_VAL(name)[1] == 'A')
-	 && (ZSTR_VAL(name)[2] == 's' || ZSTR_VAL(name)[2] == 'S')
-	 && (ZSTR_VAL(name)[3] == 't' || ZSTR_VAL(name)[3] == 'T')) {
-		return (zend_function*)&zend_ffi_cast_fn;
-	} else if (ZSTR_LEN(name) == sizeof("type") -1
-	 && (ZSTR_VAL(name)[0] == 't' || ZSTR_VAL(name)[0] == 'T')
-	 && (ZSTR_VAL(name)[1] == 'y' || ZSTR_VAL(name)[1] == 'Y')
-	 && (ZSTR_VAL(name)[2] == 'p' || ZSTR_VAL(name)[2] == 'P')
-	 && (ZSTR_VAL(name)[3] == 'e' || ZSTR_VAL(name)[3] == 'E')) {
-		return (zend_function*)&zend_ffi_type_fn;
+	zv = zend_hash_find(&zend_ffi_scope_ce->function_table, name);
```

Comment:
Now you are not handling lowercase names, for example this code used to work but now will not anymore:
```php
<?php
$scope = FFI::cdef("");
$x = $scope->TYPE("struct X { int x; }"); // TYPE in uppercase instead of lowercase used to work
var_dump($x);
```

Also, I think `zend_hash_find_ptr` or `zend_hash_find_ptr_lc` directly into a `zend_function *` would be cleaner code.

Reply:
> Now you are not handling lowercase names, for example this code used to work but now will not anymore:

Thanks for catching!

> Also, I think `zend_hash_find_ptr` or `zend_hash_find_ptr_lc` directly into a `zend_function *` variable would be cleaner code.

Cleaner, but one check more.

Reply:
> Cleaner, but one check more.

Ok, that is fair.

Reply:
Fixed through zend_hash_find_ptr_lc() usage